### PR TITLE
Cp agp iteration one

### PR DESCRIPTION
**Pull request by agpiermarini**
Continued work on iteration 1. Request to merge through standard deviation methods.

### CODE REVIEW

File: lib/item_repo.rb [hunk 1]
```diff
@@ -2,11 +2,12 @@
...
-  attr_reader :items
-  def initialize(csv)
-    @items = []
+  attr_reader :items, :parent
+  def initialize(csv, parent)
+    @items  = []
+    @parent = parent
     DataAnalyst.find_items(csv).each do |item|
-      @items << Item.new(item[0], item[1], item[2], item[3], item[4], item[5], item[6])
+      @items << Item.new(item[0], item[1], item[2], item[3], item[4], item[5], item[6], self)
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/item_repo.rb:10]: Will refactor later if necessary.

File: lib/item.rb [hunk 2]
```diff
@@ -1,17 +1,22 @@
 # This is the Item class
 class Item
-  attr_reader :id, :name, :description, :unit_price, :merchant_id, :created_at, :updated_at
-  def initialize(id, name, description, unit_price, merchant_id, created_at, updated_at)
+  attr_reader :id, :name, :description, :unit_price, :merchant_id, :created_at, :updated_at, :parent
+  def initialize(id, name, description, unit_price, merchant_id, created_at, updated_at, parent)
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/item.rb:4]: Will refactor later if necessary.

File: lib/item.rb [hunk 3]
```diff
@@ -1,17 +1,22 @@
 # This is the Item class
 class Item
-  attr_reader :id, :name, :description, :unit_price, :merchant_id, :created_at, :updated_at
-  def initialize(id, name, description, unit_price, merchant_id, created_at, updated_at)
+  attr_reader :id, :name, :description, :unit_price, :merchant_id, :created_at, :updated_at, :parent
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/item.rb:3]: Will refactor later if necessary.

File: test/sales_analyst_test.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  def test_existence
+    sa = SalesAnalyst.new(@se)
+
+    assert_instance_of SalesAnalyst, sa
+  end
+
+  def test_average_items_per_merchant
+    sa = SalesAnalyst.new(@se)
+
+    assert_equal (@se.items.all.size / @se.merchants.all.size), sa.average_items_per_merchant
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply test/sales_analyst_test.rb]: Will refactor later if necessary.

File: lib/item_repo.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/item_repo.rb:10]: Will refactor later if necessary.

File: lib/item.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/item.rb:4]: Will refactor later if necessary.

File: lib/item.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (agpiermarini) [reply lib/item.rb:3]: Will refactor later if necessary.

File: test/item_test.rb [hunk 5]
```diff
@@ -19,7 +19,7 @@ def test_attributes
   end
 
   def test_price_to_dollars
-    item = Item.new(1, 'Pencil', 'You can use it to write things', BigDecimal.new(10.99,4), 1, Time.now, Time.now)
+    item = Item.new(1, 'Pencil', 'You can use it to write things', BigDecimal.new(10.99, 4), 1, Time.now, Time.now, nil)
```
[the remark this answers is not in the document]
Comment (agpiermarini) [reply test/item_test.rb:22]: Will refactor later if necessary.

File: test/item_test.rb [hunk 6]
```diff
@@ -4,13 +4,13 @@
...
   def test_existence
-    item = Item.new(1,'Pencil', 'You can use it to write things', BigDecimal.new(10.99,4), 1, Time.now, Time.now)
+    item = Item.new(1, 'Pencil', 'You can use it to write things', BigDecimal.new(10.99, 4), 1, Time.now, Time.now, nil)
 
     assert_instance_of Item, item
   end
 
   def test_attributes
-    item = Item.new(1,'Pencil', 'You can use it to write things', BigDecimal.new(10.99,4), 1, Time.now, Time.now)
+    item = Item.new(1, 'Pencil', 'You can use it to write things', BigDecimal.new(10.99, 4), 1, Time.now, Time.now, nil)
```
[the remark this answers is not in the document]
Comment (CaptainSpectacular) [reply test/item_test.rb:13]: Will refac when doing cleanup.

File: test/item_test.rb [hunk 7]
```diff
@@ -4,13 +4,13 @@
 
 class ItemTest < MiniTest::Test
   def test_existence
-    item = Item.new(1,'Pencil', 'You can use it to write things', BigDecimal.new(10.99,4), 1, Time.now, Time.now)
+    item = Item.new(1, 'Pencil', 'You can use it to write things', BigDecimal.new(10.99, 4), 1, Time.now, Time.now, nil)
```
[the remark this answers is not in the document]
Comment (CaptainSpectacular) [reply test/item_test.rb:7]: Will refac when doing cleanup.

**Comment by CaptainSpectacular** [comment]
Approved

**Pull request merged**